first draft

eea-edo opened the pull request.

Commit:
first draft

Jan-Marcek commented on src/main/java/org/opendatanode/plugins/transformer/exceltocsv/ExcelToCsv.java:
```diff
@@ -0,0 +1,263 @@
+                Row row = sheet.getRow(r);
+                if (row != null) {
+                    List<String> csvLine = new ArrayList<>();
+                    for (int c = minCellNumber; c < row.getLastCellNum(); c++) {
```
Don't need to loop over all items in list. When we find first non-empty cell then we can stop.

Jan-Marcek commented on src/main/java/org/opendatanode/plugins/transformer/exceltocsv/ExcelToCsv.java:
```diff
@@ -0,0 +1,263 @@
+        return parts[0];
+    }
+
+    private void addFileToOutput(DpuFile file) throws DataUnitException {
```
an unused method.

Jan-Marcek commented on src/main/java/org/opendatanode/plugins/transformer/exceltocsv/ExcelToCsv.java:
```diff
@@ -0,0 +1,263 @@
+            throw ex;
+        } finally {
+            if (wb != null) {
+                wb.close();
```
IntelliJ says "IOException' might be thrown inside 'finally' "

Jan-Marcek commented on src/main/java/org/opendatanode/plugins/transformer/exceltocsv/ExcelToCsv.java:
```diff
@@ -0,0 +1,263 @@
+            }
+        } finally {
+            if (csvWriter != null) {
+                csvWriter.close();
```
IntelliJ says "IOException' might be thrown inside 'finally' "

Commit:
Updated excel example for better testing.

Commit:
Optimization of finding min. cell and fix of exception handling.

eea-edo committed:
Update README.md

eea-edo committed:
Update README.md

Commit:
added ExcelToCsvUtils class that contains util methods from ExcelToCsv class.

hanecak committed:
fixed minor typos

hanecak committed:
ignore target directory and .eclipse setting directory

hanecak committed:
process workbooks in read-only mode to avoid write problems + poi-ooxml updated from 3.11 to 3.14 to incorporate read-noly bugfixes

While working with some data from City of Presov (XLSX file at http://egov.presov.sk/Default.aspx?NavigationState=180:0::plac240:_144117_5_32
referenced at https://data.gov.sk/dataset/http-egov-presov-sk-default-aspx-navigationstate-180-0-plac240-144117-5-1/resource/dcf508e1-37cf-4bb6-954d-7d3f7819df95
following error was observed:

Caused by: org.apache.poi.openxml4j.exceptions.OpenXML4JRuntimeException: Fail to save: an error occurs while saving the package : part
at org.apache.poi.openxml4j.opc.ZipPackage.saveImpl(ZipPackage.java:503)
at org.apache.poi.openxml4j.opc.OPCPackage.save(OPCPackage.java:1425)
at org.apache.poi.openxml4j.opc.OPCPackage.save(OPCPackage.java:1412)
at org.apache.poi.openxml4j.opc.ZipPackage.closeImpl(ZipPackage.java:353)
at org.apache.poi.openxml4j.opc.OPCPackage.close(OPCPackage.java:425)
at org.apache.poi.POIXMLDocument.close(POIXMLDocument.java:179)
at org.apache.poi.xssf.usermodel.XSSFWorkbook.close(XSSFWorkbook.java:1463)
at org.opendatanode.plugins.transformer.exceltocsv.ExcelToCsv.excelToCsv(ExcelToCsv.java:117)
at org.opendatanode.plugins.transformer.exceltocsv.ExcelToCsv.innerExecute(ExcelToCsv.java:90)
at eu.unifiedviews.helpers.dpu.exec.AbstractDpu.execute(AbstractDpu.java:133)
... 4 more

hanecak commented:
Depends on https://github.com/UnifiedViews/Plugins/pull/389 and https://github.com/UnifiedViews/Core/pull/594 .

hanecak commented:
I've moved the DPU directly into UnifiedViews/Plugins repository, see https://github.com/UnifiedViews/Plugins/pull/391 .

Thus this particular pull request is no longer needed.

Later, I will also delete this repository.
